staking-miner: CLI flag delay solution x secs

niklasad1 opened the pull request.

niklasad1 commented on utils/staking-miner/src/monitor.rs:
```diff
@@ -198,6 +198,7 @@ macro_rules! monitor_cmd_for { ($runtime:tt) => { paste::paste! {
 				return;
 			}
 
+
```
```suggestion
```

kianenigma commented:
Can you also double check that: 

1. We do read the status of the _UPDATED_ queue before submitting, 
2. and that if we observe a solution in it with the same score as what we have submitted, 
3. or if it is actually full
4. we emit a log, and we don't submit?

niklasad1 commented:
Yeah, we check that the solution is better than the existing ones after mining but we doesn't emit logs for that.

On other hand we don't check if the queue is full as we only submit if we are better but why should we?

niklasad1 commented:
> Can you also double check that: 

> 1. We do read the status of the _UPDATED_ queue before submitting, 
> 2. and that if we observe a solution in it with the same score as what we have submitted, 
> 3. or if it is actually full
> 4. we emit a log, and we don't submit? 

1. Yes, that has been done all the time but I added that we read that `current account` hasn't submitted in that round as well.
2.  Added a log if that occurs but as `if-leading stategy` is used by default nothing will be sent if the scores as equal, as we compare the scores with `ElectionScore::strict_better_better(other, episilon)`
3. If it's better we will still submit but added a log if the queue is full.
4. Yes added.

dvdplm commented on utils/staking-miner/src/opts.rs:
```diff
@@ -83,6 +83,17 @@ pub(crate) struct MonitorConfig {
+	///
+	/// At every block when a solution is attempted
+	/// a delay can be enforced to avoid submitting at
+	/// "same time" and risk potentially races to other miners.
```
```suggestion
	/// "same time" and risk potential races with other miners.
```

dvdplm commented on utils/staking-miner/src/opts.rs:
```diff
@@ -83,6 +83,17 @@ pub(crate) struct MonitorConfig {
+	/// a delay can be enforced to avoid submitting at
+	/// "same time" and risk potentially races to other miners.
+	///
+	/// This may mean if there are competing solutions you might not submit
+	/// if the scores are equal once this is enabled.
```
```suggestion
	/// When this is enabled and there are competing solutions, your solution might not be submitted
	/// if the scores are equal.
```

niklasad1 commented:
bot merge
